src/wmtk/TriMeshOperationExecutor.cpp

```diff
@@ -78,6 +78,12 @@ TriMesh::TriMeshOperationExecutor::TriMeshOperationExecutor(
         m_incident_face_datas.emplace_back(get_incident_face_data(f.tuple()));
     }
 
+    assert(m_incident_face_datas.size() <= 2);
+    if (m_incident_face_datas[0].fid != m.id_face(m_operating_tuple)) {
+        assert(m_incident_face_datas.size() == 2);
+        std::swap(m_incident_face_datas[0], m_incident_face_datas[1]);
+    }
```

Comment:
LGTM. According to my understanding, the reason for performing this swap is that when computing the `closed_star`, the returned faces are sorted by their IDs. However, since we may need to revert to the original order later on, why do we still use` closed_star` to retrieve `m_incident_face_datas`?

Reply:
This was meant as a generalization for the 3d case. It is true that in this case, just requesting the two neighbors would have been easier.